src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1649,6 +1652,45 @@ private void DeleteMultipleCertificates(TransactionOperationContext context, str
             }
         }
 
+        private static void DeleteDatabaseFromCertificatePermissions(TransactionOperationContext context, string database)
+        {
+            try
+            {
+                var certTable = context.Transaction.InnerTransaction.OpenTable(CertificatesSchema, CertificatesSlice);
+
+                foreach (var result in certTable.SeekByPrimaryKeyPrefix(Slices.Empty, Slices.Empty, 0))
+                {
+                    var blittable = GetCertificate(context, result.Value).Cert;
+
+                    if (blittable.TryGet(nameof(CertificateDefinition.Thumbprint), out string thumbprint) == false)
+                        throw new MissingFieldException($"Couldn't get '{nameof(CertificateDefinition.Thumbprint)}' from {nameof(CertificateDefinition)}");
+                    
+                    if (blittable.TryGet(nameof(CertificateDefinition.PublicKeyPinningHash), out string hash) == false)
+                        throw new MissingFieldException($"Couldn't get '{nameof(CertificateDefinition.PublicKeyPinningHash)}' from {nameof(CertificateDefinition)}");
+                    
+                    if (blittable.TryGet(nameof(CertificateDefinition.Permissions), out BlittableJsonReaderObject permissions) == false)
+                        throw new MissingFieldException($"Couldn't get '{nameof(CertificateDefinition.Permissions)}' from {nameof(CertificateDefinition)}");
+
+                    using (Slice.From(context.Allocator, thumbprint.ToLower(), out var thumbprintSlice))
+                    using (Slice.From(context.Allocator, hash, out var hashSlice))
+                    {
+                        int index = permissions.GetPropertyIndex(database);
+                        if (index > -1)
+                        {
+                            permissions.Modifications ??= new DynamicJsonValue();
+                            permissions.Modifications.Removals = new HashSet<int>() { index };
```

Comment:
```suggestion
                            permissions.Modifications.Remove(database);
```

Reply:
Let's avoid using internal details of the value

Reply:
Done